### ISSUE TITLE
Support for custom blocks

### ISSUE THREAD

**Issue by dswd** [Keats/tera#127]
I am currently investigating building a static site generator based on tera and it would be great to have custom blocks that transform their contents like 
```
{% markdown %}
Some
Markdown
formatting
{% end markdown %}
```

It would be great if tera was extensible in this way and allowed people to add their own blocks. Maybe this could also be realized with #102 as `{% filter markdown %}...{% endfilter %}`.

Currently the only way to realize this functionality is a pain in the ***:
```
{{ "Some\nmarkdown\nformatting" | markdown }}
```

**Comment by Keats** [comment]
Filter sections and custom blocks are different things.
Filter sections operate only on strings and return a string, like a filter. This is what you need right now I believe.
Custom blocks would need access to the AST and handle the rendering themselves, and I'm not sure how to implement that yet!

Filter sections are easy to implement: a new node in the AST with a name (the filter used) and a body. The renderer will first render the body and then apply the filter to the body string.

I'm not sure if I will have the time to do it myself this week, but that would be a good first PR for someone looking to contribute to Tera!

**Comment by Keats** [comment]
Your patch has been released in 0.7.1, thanks!
